lib/routes/profile.js

```diff
@@ -43,27 +56,39 @@ module.exports = {
     }
   },
   handler: function email(req, reply) {
+    const server = req.server;
     var creds = req.auth.credentials;
     if (!hasAllowedScope(creds.scope || [])) {
       return reply(Boom.forbidden());
     }
-    batch(req, {
-      email: '/v1/email',
-      uid: '/v1/uid',
-      avatar: '/v1/avatar',
-      displayName: '/v1/display_name'
-    })
-    .done(function (result) {
-      if (creds.scope.indexOf('openid') !== -1) {
-        result.sub = creds.user;
-      }
-      var rep = reply(result);
-      var etag = computeEtag(result);
-      if (etag) {
-        rep = rep.etag(etag);
+
+    if (!server.methods.batch) {
+      createServerMethod(server);
```

Comment:
Is there any way to clean this up a bit. Inside of the `createServerMethod` we have `req.auth.credentials.user`. It looks a bit confusing because of a mix of a request scope and some global scope via `if (!server.methods.batch) {`